openstacksdk_plugin/resources/compute/server.py

```diff
@@ -32,6 +32,74 @@
                                            SERVER_TASK_STOP,)
 
 
+def _set_server_ips_runtime_properties(server):
+    addresses = server.addresses
+    if not addresses:
+        return None
+
+    ipv4_addresses = []
+    ipv6_addresses = []
+
+    for network, addresses in addresses.iteritems():
+        for address in addresses:
+            # ip config
+            ipv4 = dict()
+            ipv4['addr'] = address['addr']
+            ipv4['type'] = address['OS-EXT-IPS:type']
+
+            # Check where `ip_config` should be added
+            if address['version'] == 4:
+                ipv4_addresses.append(ipv4)
+            elif address['version'] == 6:
+                ipv6_addresses.append(address['addr'])
+
+    # Check if access_ipv4 is set or not
+    if server.access_ipv4:
+        ctx.instance.runtime_properties['access_ipv4'] = server.access_ipv4
+
+    # Check if access_ipv6 is set or not
+    if server.access_ipv6:
+        ctx.instance.runtime_properties['access_ipv6'] = server.access_ipv6
+
+    # If "ipv4_addresses" is only contains one item, them we need to check
+    # both private/public ip in order to set them as part of runtime_properties
+    for ipv4 in ipv4_addresses:
+        ip = ipv4['addr']
+
+        # Only set the first "ip" as runtime property
+        if ipv4['type'] == 'fixed'\
+                and 'ip' not in ctx.instance.runtime_properties:
+            ctx.instance.runtime_properties['ip'] = ip
+
+        # Only set the first "public_ip_address" as runtime property
+        elif ipv4['type'] == 'floating'\
+                and 'public_ip_address' not in ctx.instance.runtime_properties:
+            ctx.instance.runtime_properties['public_ip_address'] = ip
+
+    for ipv6 in ipv6_addresses:
```

Comment:
We should probably run this before ipv4

Reply:
Can you explain more ? why should we do it

Reply:
If `ip` is set in the loop of `ipv4_addresses` and the we try to set it in `ipv6_addresses`, the correct value will be overwritten by an ipv6 address.

Reply:
I think at both order this issue will happen so I'm going to change the variable name and use another one instead

Reply:
```
    for ipv6 in ipv6_addresses:
        ip_v6 = ipv6['addr']

        # Only set the first "ipv6" as runtime property
        if ipv6['type'] == 'fixed' \
                and 'ipv6' not in ctx.instance.runtime_properties:
            ctx.instance.runtime_properties['ipv6'] = ip_v6

        # Only set the first "public_ip6_address" as runtime property
        elif ipv6['type'] == 'floating'\
                and 'public_ip6_address' not in\
                    ctx.instance.runtime_properties:
            ctx.instance.runtime_properties['public_ip6_address'] = ip_v6
```